### ISSUE TITLE
How to change LED Color

### ISSUE THREAD

**Issue by Collonville** [HipsterSloth/psmove-unity5#20]
I`m using psmove-unity5 to proceed my VR project.I want to know how to change the LED color attached on the PSMove with the script(for a example : Square button pressed -> turn to red)
I use unity 5.3.3 and windows 10 64bit.

**Comment by HipsterSloth** [comment]
There are two ways to set the tracking color currently:

1. Set the InitialTrackingColor on the [PSMoveManager](https://github.com/HipsterSloth/psmove-unity5/blob/master/Assets/psmove-unity5/PSMoveManager.cs#L58). But this just sets the default tracking color at start-up
1. If you want to cycle the tracking color to the next available tracking color (magenta -> cyan ->  use [PSMoveController.CycleTrackingColor](https://github.com/HipsterSloth/psmove-unity5/blob/master/Assets/psmove-unity5/PSMoveController.cs#L144). This will restart the controller blinking color. 

If you want to set an arbitrary color, there is an API in PSMoveAPI to set an arbitrary color, but that's not exposed in PSMove-Unity5. However it would be very easy to add. Is that what you would like?

**Comment by Collonville** [comment]
Yes, I want use arbitrary color in psmove-unity5.There is psmove_set_leds() in PSMoveAPI but is it possible to use it?

**Comment by HipsterSloth** [comment]
Yes with some caveats. 

It'd be pretty straightforward to expose that API to psmove-unity5. An entry needs to be added in [PSMoveAPI.cs](https://github.com/HipsterSloth/psmove-unity5/blob/master/Assets/psmove-unity5/PSMoveAPI.cs#L361-L362) to expose the set leds method, and then we need to add a color request similar to what we do for color cycling in (PSMoveData.cs)[https://github.com/HipsterSloth/psmove-unity5/blob/master/Assets/psmove-unity5/PSMoveData.cs#L56]. The PSMoveManager's worker thread will then pick up this request and then call the exposed psmove_set_leds api call,

But if you want to maintain tracking you would have to do the blink calibration again. We have to do the blink calibration because the color you set doesn't necessarily match the color the camera sees (due to camera exposure settings and what not). If you don't care about tracking and just want to set the color that's a bit easier. It can just be set immediately. Since you mentioned this is for a VR project I assume you need the controller's position to be tracked (not just orientation which doesn't use camera tracking)?  

Also is there a reason that the color has to be switched in the middle of runtime rather than at the beginning like it is now? Typically you only switch colors if the one you are using isn't tracking well due to local lighting conditions. I'm not opposed to adding switching to an arbitrary color, but I'm curious what your use-case is exactly.
